pkg/auth/dependency/principal/password/login_id_checker.go

```diff
@@ -0,0 +1,160 @@
+package password
+
+import (
+	"strings"
+	"unicode"
+
+	"github.com/skygeario/skygear-server/pkg/core/auth/metadata"
+	"github.com/skygeario/skygear-server/pkg/core/config"
+	"github.com/skygeario/skygear-server/pkg/core/validation"
+)
+
+type LoginIDTypeChecker interface {
+	Validate(loginID string) error
+}
+
+type LoginIDTypeCheckerFactory interface {
+	NewChecker(loginIDKey string) LoginIDTypeChecker
```

Comment:
I think we can directly pass in a `LoginIDKeyConfiguration`, instead of looking it up twice.

Reply:
Do you mean pass in the `LoginIDKeyType`? Yes, that would be better